src/web_app/js/util.js

```diff
@@ -118,15 +107,27 @@ function parseJSON(json) {
 }
 
 // Filter a list of TURN urls to only contain those with transport=|protocol|.
-function filterTurnUrls(urls, protocol) {
-  for (var i = 0; i < urls.length;) {
-    var parts = urls[i].split('?');
-    if (parts.length > 1 && parts[1] !== ('transport=' + protocol)) {
-      urls.splice(i, 1);
-    } else {
-      ++i;
+function filterTurnUrls(config, protocol) {
```

Comment:
This function leaves the stun servers if they specify the requested protocol, but does not modifies an unspecified stun address to add the protocol. It feels incoherent to me.

Suggestion 1: rename to filterIceServersUrls and treat all urls the same independently of startsWith.
Suggestion 2: filter out all urls that do not startwith('turn') even if they specific the right transport.

The reason why this needed to be re-written is that this code was creating an iceServer entry with no urls since networktraversal reports stun server aside from turn servers.